go/adbc/driver/snowflake/driver_test.go

```diff
@@ -467,7 +459,7 @@ func (suite *SnowflakeTests) TestSqlIngestRecordAndStreamAreEquivalent() {
 		},
 	}, nil)
 
-	bldr := array.NewRecordBuilder(mem, sc)
+	bldr := array.NewRecordBuilder(suite.Quirks.Alloc(), sc)
```

Comment:
do we properly validate the Quirks.Alloc with a checked allocator after tests?

Reply:
should we create a scoped / checked allocator from `suite.Quirks.Alloc()`?

Reply:
Same answer as above. By using [Setup/TeardownDriver](https://github.com/apache/arrow-adbc/blob/main/go/adbc/driver/snowflake/driver_test.go#L59-L71) we are creating and asserting a new CheckedAllocator for every test.